### PR TITLE
Add `GLVParameters` trait definition

### PR DESCRIPTION
**Pull request by zhenfeizhang**
## Description

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (master)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work: _This PR defines `GLV parameters` trait for `ark-ec` and will be useful in `bandersnatch` implementation_
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

not related
- [ ] Wrote unit tests

### CODE REVIEW

**Comment by zhenfeizhang** [comment]
Note that this GLV parameters use a different set of definitions, and will not be compatible with a draft definition in  https://github.com/arkworks-rs/algebra/pull/127 (which seems to be inactive)

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## Pending
 
+- [\#301](https://github.com/arkworks-rs/algebra/pull/201) (ark-ec) Add `GLVParameters` trait definition.
```
Comment (swasilyev) [inline CHANGELOG.md:5]: should be https://github.com/arkworks-rs/algebra/pull/301

Comment (zhenfeizhang) [reply CHANGELOG.md]: Thanks, and fixed.

**Comment by Pratyush** [comment]
Hey @zhenfeizhang , thanks for the PR! I'll try to review this within the next few days!

File: ec/src/glv.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+/// The GLV parameters that are useful to compute the endomorphism
+/// and scalar decomposition.
+pub trait GLVParameters: Send + Sync + 'static + ModelParameters {
+    type CurveAffine;
+    type CurveProjective;
+
+    // phi(P) = lambda*P for all P
+    // constants that are used to calculate phi(P)
+    const COEFF_A1: Self::BaseField;
```
Comment (ValarDragon) [inline ec/src/glv.rs:11]: Can there be a docstring explaining what is A1, A2, A3, B1, B2 etc?

Comment (zhenfeizhang) [reply ec/src/glv.rs:11]: Will do!

Comment (zhenfeizhang) [reply ec/src/glv.rs:13]: The docstrings are added

**Review by weikengchen — APPROVED** [review]

**Pull request merged**